### PR TITLE
dev/core#1179 Default Contact Search Profile should be respected for contact searches including Edit Smart Group Criteria

### PR DESCRIPTION
**Pull request by sushantpaste**
https://lab.civicrm.org/dev/core/issues/1179

Default Contact Search Profile should be respected for contact searches including Edit Smart Group Criteria.

* Create a profile configured to be used in Search Views.
* Make this profile as Default Contact Search Profile. From Administrator - Search Preferences - Default Contact Search Profile.
* Go to Basic, Advanced Search/Builer, all the results have profile fields listed as column headers if use force=1
* Create a smart search and go to Contacts and use force=1. Profile fields not shown.

Before:

before-patch-view

After:

Smart group advance search will also show the default Search profile fields.

after-patch-view

### CODE REVIEW

**Comment by yashodha** [comment]
@monishdeb @mlutfy 

Can you please QA this?

**Comment by eileenmcnaughton** [comment]
@sunilpawar can you please fill in the PR template? Although it DOES duplicate gitlab details the PR templates are the primary source for the information for the release notes (it also makes it likelier someone will review it if the PR is easy to understand with a quick look)

**Comment by yashodha** [comment]
@eileenmcnaughton you mean @sushantpaste :)

**Comment by eileenmcnaughton** [comment]
dang autocomplete ....

**Comment by sushantpaste** [comment]
@eileenmcnaughton @yashodha Updated the issues information

**Comment by yashodha** [comment]
@irfanhaq1  @monishdeb @JoeMurray @mlutfy 
Can you please QA?

**Comment by monishdeb** [comment]
I am ok with the patch. But wondering if we can capture the behavior in a unit test.

File: CRM/Contact/Form/Search.php [hunk 1]
```diff
@@ -612,26 +612,27 @@ public function preProcess() {
           $this->_ssID = $ssId;
         }
       }
-
+      // set the uf group id if not already present
+      if (isset($this->_ufGroupID)) {
```
Comment (mlutfy) [inline CRM/Contact/Form/Search.php]: The comment contradicts the code. I would recommend describing the use-case that this is fixing.

**Comment by mlutfy** [comment]
Looks OK to me, but agree with Monish. Looking only at the code, it's the kind of feature that could easily be lost by someone else doing another change. Having more verbose comments or a test could avoid regressions.

Side note: we should avoid a "same shop merge", i.e. this PR is for a specific client which many of us are working with (except Eileen).

**Comment by sushantpaste** [comment]
@JoeMurray @yashodha  Fixed comment. @eileenmcnaughton This also needs test any help from you please.

**Comment by eileenmcnaughton** [comment]
@sushantpaste ok great - let's focus on the other one first...

**Comment by JoeMurray** [comment]
So this seems mergeable now.

**Comment by eileenmcnaughton** [comment]
I think it's just the test - @sushantpaste had another PR he was going to tackle a test for first

**Comment by mlutfy** [comment]
Any update on this PR? Is it still blocked by a test?

**Comment by eileenmcnaughton** [comment]
It's also stale so let's close for now - please re-open once it's re-based (ideally with a test to get review)

**Pull request closed without merging**